libs/ml/include/ml/dataloaders/word2vec_loaders/w2v_dataloader.hpp

```diff
@@ -17,17 +17,15 @@
 //
 //------------------------------------------------------------------------------
 
+#include "ml/dataloaders/word2vec_loaders/w2v_dataloader.hpp"
+
 #include "core/random/lcg.hpp"
 #include "math/tensor/tensor.hpp"
 #include "meta/type_traits.hpp"
 #include "ml/dataloaders/dataloader.hpp"
```

Comment:
most of these includes can be moved into source and just forward declare the types you need for the header i think